File: backend/src/models/shipment.js
import mongoose from "mongoose";

//Schema for a Shipment
const shipmentSchema = new mongoose.Schema({
  trackingNumber: String,
  street: String,
  city: String,
  zip: String,
  country: String,
  status: String,
  weight: String,
});

export const Shipment = mongoose.model("Shipment", shipmentSchema);

File: backend/models/shipment.go
package models

import (
	"gorm.io/gorm"
)

type Shipment struct {
	gorm.Model
	TrackingNumber string `json:"trackingNumber"`
	Street         string `json:"street"`
	City           string `json:"city"`
	Zip            string `json:"zip"`
	Country        string `json:"country"`
	Status         string `json:"status"`
	Weight         string `json:"weight"`
}
